templates/vue-demo-store/app/pages/account/index.vue

```diff
@@ -50,16 +53,15 @@ const updateNewsletterStatus = async () => {
     console.log("error", error);
     pushError(t("messages.error"));
   } finally {
-    getNewsletterStatus().then(() => {
-      newsletter.value = isNewsletterSubscriber.value;
-    });
+    await getNewsletterStatus();
```

Comment:
shouldn't that be in another method? We can also have error with this endpoint and then the field stays disabled with no indication why